File: src/android/com/adobe/phonegap/push/FullScreenActivity.java
package com.adobe.phonegap.push;

import android.app.Activity;
import android.app.KeyguardManager;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.util.Log;
import android.view.WindowManager;

public class FullScreenActivity extends Activity implements PushConstants {
    private static final String LOG_TAG = "FullScreenActivity";

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        Log.d(LOG_TAG, "onCreate");
        super.onCreate(savedInstanceState);

        turnScreenOnAndKeyguardOff();
        forceMainActivityReload();
        finish();
    }

    private void forceMainActivityReload () {
        PackageManager pm = getPackageManager();
        Intent launchIntent = pm.getLaunchIntentForPackage(getApplicationContext().getPackageName());

        launchIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        launchIntent.addFlags(Intent.FLAG_FROM_BACKGROUND);

        startActivity(launchIntent);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        turnScreenOffAndKeyguardOn();
    }

    private void turnScreenOnAndKeyguardOff() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
            Log.d(LOG_TAG, "setShowWhenLocked");
            setShowWhenLocked(true);
            setTurnScreenOn(true);
        } else {
            Log.d(LOG_TAG, "addFlags");
            getWindow().addFlags(
                WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
                | WindowManager.LayoutParams.FLAG_ALLOW_LOCK_WHILE_SCREEN_ON
            );
        }
        Object candidate = getSystemService(Context.KEYGUARD_SERVICE);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && candidate != null) {
            KeyguardManager keyguardManager = (KeyguardManager) candidate;
            keyguardManager.requestDismissKeyguard(this, new KeyguardManager.KeyguardDismissCallback() {
                @Override
                public void onDismissError() {
                    super.onDismissError();
                    Log.d(LOG_TAG, "onDismissError");
                }

                @Override
                public void onDismissSucceeded() {
                    super.onDismissSucceeded();
                    Log.d(LOG_TAG, "onDismissSucceeded");
                }
            });
        }
    }

    private void turnScreenOffAndKeyguardOn() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
            setShowWhenLocked(false);
            setTurnScreenOn(false);
        } else {
            getWindow().clearFlags(
                WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
                | WindowManager.LayoutParams.FLAG_ALLOW_LOCK_WHILE_SCREEN_ON
            );
        }
    }
}

File: src/android/com/adobe/phonegap/push/FullScreenActivity.kt
package com.adobe.phonegap.push

import android.app.Activity
import android.app.KeyguardManager
import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.util.Log
import android.view.WindowManager

class FullScreenActivity : Activity(), PushConstants {
    @Override
    fun onCreate(@Nullable savedInstanceState: Bundle?) {
        Log.d(LOG_TAG, "onCreate")
        super.onCreate(savedInstanceState)
        turnScreenOnAndKeyguardOff()
        forceMainActivityReload()
        finish()
    }

    private fun forceMainActivityReload() {
        val pm: PackageManager = getPackageManager()
        val launchIntent: Intent =
            pm.getLaunchIntentForPackage(getApplicationContext().getPackageName())
        launchIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
        launchIntent.addFlags(Intent.FLAG_FROM_BACKGROUND)
        startActivity(launchIntent)
    }

    @Override
    protected fun onDestroy() {
        super.onDestroy()
        turnScreenOffAndKeyguardOn()
    }

    private fun turnScreenOnAndKeyguardOff() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
            Log.d(LOG_TAG, "setShowWhenLocked")
            setShowWhenLocked(true)
            setTurnScreenOn(true)
        } else {
            Log.d(LOG_TAG, "addFlags")
            getWindow().addFlags(
                WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
                        or WindowManager.LayoutParams.FLAG_ALLOW_LOCK_WHILE_SCREEN_ON
            )
        }
        val candidate: Object = getSystemService(Context.KEYGUARD_SERVICE)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && candidate != null) {
            val keyguardManager: KeyguardManager = candidate as KeyguardManager
            keyguardManager.requestDismissKeyguard(this, object : KeyguardDismissCallback() {
                @Override
                fun onDismissError() {
                    super.onDismissError()
                    Log.d(LOG_TAG, "onDismissError")
                }

                @Override
                fun onDismissSucceeded() {
                    super.onDismissSucceeded()
                    Log.d(LOG_TAG, "onDismissSucceeded")
                }
            })
        }
    }

    private fun turnScreenOffAndKeyguardOn() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
            setShowWhenLocked(false)
            setTurnScreenOn(false)
        } else {
            getWindow().clearFlags(
                WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
                        or WindowManager.LayoutParams.FLAG_ALLOW_LOCK_WHILE_SCREEN_ON
            )
        }
    }

    companion object {
        private const val LOG_TAG = "FullScreenActivity"
    }
}
